tests/TCP_echo_client.py

```diff
@@ -286,33 +298,39 @@ def run(self):
                         else:
                             pass  # logger.log("DEBUG: ignoring EVENT_WRITE")
 
-            # delay the close if necessary
-            while self.delay_close:
+            # If delaying the close of the connection block until wait is
+            # called
+            if self.delay_close:
                 self.logger.log("%s Delayed closing wait..." % self.prefix)
-                self.is_waiting_to_close = True
-                self._lock.acquire()
-                self._cond.wait(self.timeout)
-                self._lock.release()
+                while self._delay_waiting.wait(timeout=TIMEOUT) is False:
```

Comment:
Can this be a if statement instead of a while statement ?
```  
if self._delay_waiting.wait(timeout=TIMEOUT) is False:
    pass
```

Reply:
I could... but if it throws a timeout is that an error?  It's basically done (successfully) and just waiting for the test to reclaim it.  Removing the loop means we cannot use the standalone client to simply hold TCP connections open in the background while we execute long running tests.

But I suppose the other argument is that it can possibly hang forever if the main test hangs...  dunno.